src/pip/_internal/index.py

```diff
@@ -802,6 +802,14 @@ def get_page(cls, link, skip_archives=True, session=None):
                 url = urllib_parse.urljoin(url, 'index.html')
                 logger.debug(' file: URL is directory, getting %s', url)
 
+            if scheme != 'file' and netloc in session.blacklisted_hosts:
+                logger.debug(
+                    'Skipping page %s because host %s is blacklisted',
```

Comment:
I'm afraid this `blacklisted` message will be unclear to users.
Maybe: `'Skipping page %s because host %s is ignored due to a connection error'`

Reply:
👍